packages/web/src/ssr/track/onRenderHtml.tsx

```diff
@@ -0,0 +1,98 @@
+import { SsrPageProps } from '@audius/common/models'
+import createEmotionServer from '@emotion/server/create-instance'
+import { createMemoryHistory } from 'history'
+import { renderToString } from 'react-dom/server'
+import { Helmet } from 'react-helmet'
+import { escapeInject, dangerouslySkipEscape } from 'vike/server'
+import type { PageContextServer } from 'vike/types'
+
+import { ServerWebPlayer } from 'app/web-player/ServerWebPlayer'
+import { ServerTrackPage } from 'pages/server-track-page/ServerTrackPage'
+import { isMobileUserAgent } from 'utils/clientUtil'
+import { getTrackPageSEOFields } from 'utils/seo'
+
+import { harmonyCache } from '../../HarmonyCacheProvider'
+import { getIndexHtml } from '../getIndexHtml'
+
+const { extractCriticalToChunks, constructStyleTagsFromChunks } =
+  createEmotionServer(harmonyCache)
+
+export default function render(
+  pageContext: PageContextServer & {
+    pageProps: SsrPageProps
+    userAgent: string
+  }
+) {
+  const { pageProps, urlPathname } = pageContext
+  const { track } = pageProps
+
+  const isMobile = isMobileUserAgent(pageContext.userAgent)
+
+  const history = createMemoryHistory({
+    initialEntries: [urlPathname]
+  })
+
+  // TODO: Fix this, this pulls in libs through the routes
+  const {
+    title = '',
+    description = '',
+    canonicalUrl = '',
+    structuredData
+  } = getTrackPageSEOFields({
```

Comment:
Ah interesting, yeah this makes sense. But also kinda funny because we are passing these into ServerTrackPage, which then uses Helmet to populate things in the header which we access via `helmet.title.toString()` below. Kinda roundabout but no need to change